### PR TITLE
Add editing and commenting on subannotations to inspector view

### PR DESCRIPTION
**Pull request by MichaelGoodale**
This PR has some basic changes to allow for editing and commentating of subannotations. This consists of several specific features.

- There is now an ability to change the length/position of a subannotation like VOT. This is accomplished by simply dragging on a selected subannotation in either the waveform or spectrogram view

- There is now both an excluded and note property added to all subannotations that are edited this way. The exclusion property is a simple boolean that displays an X on any excluded subannotations in the alignment viewpanel. Notes are simply a textbox

- Any changes are temporary until they are "committed", i.e. there is a button in the UI which pushes any changes made, that also provides a prompt to ask whether the user wishes to overwrite the previous data with the new data.

- Several new shortcuts have also been added, "x" to exclude/unexclude a given subannotation, "l" and "h" to select the next or previous subannotation(I was gonna use the arrow keys but JS didn't like them for whatever reason). "c" to commit changes.

There's a few changes that I might add depending on what you think is appropriate.

 I think the exclusion field definitely should be kept as is, as it makes it so much easier for users to quickly flag bad subannotations, but the notes field could be removed and replaced with some sort of "add subannotation property" button which could allow users to choose their own properties. 

Another change that maybe should be made is more server side validation about when it's okay to update the data, not sure about how perms should be handled there.

Let me know if there are any other changes you think are necessary. 

Here's a screenshot too with the new features. 
screenshot

### CODE REVIEW

File: iscan/static/iscan/query-detail/query-detail.js [hunk 1]
```diff
@@ -31,415 +31,485 @@ angular.module('queryDetail', [
...
 
-                $scope.can_view = false;
-                for (i = 0; i < $scope.user.corpus_permissions.length; i++) {
-                    if ($scope.user.corpus_permissions[i].corpus == $stateParams.corpus_id) {
-                        $scope.can_view = $scope.user.corpus_permissions[i].can_view_detail;
-                    }
+            $scope.can_view = false;
+            for (i = 0; i < $scope.user.corpus_permissions.length; i++) {
+                if ($scope.user.corpus_permissions[i].corpus == $stateParams.corpus_id) {
+                    $scope.can_view = $scope.user.corpus_permissions[i].can_view_detail;
```
Comment (mmcauliffe) [inline iscan/static/iscan/query-detail/query-detail.js]: So here I would add the flag for `$scope.can_edit` which then will disable moving of boundaries or changing any fields values.

File: iscan/static/iscan/query-detail/query-detail.js [hunk 2]
```diff
@@ -31,415 +31,485 @@ angular.module('queryDetail', [
...
+        }
+        else {
+            $scope.can_listen = false;
+            $scope.can_edit = false;
+            $scope.can_view_annotations = false;
+            $scope.can_annotate = false;
+            for (i = 0; i < $scope.user.corpus_permissions.length; i++) {
+                if ($scope.user.corpus_permissions[i].corpus == $stateParams.corpus_id) {
+                    $scope.can_listen = $scope.user.corpus_permissions[i].can_listen;
+                    $scope.can_edit = $scope.user.corpus_permissions[i].can_edit;
```
Comment (mmcauliffe) [inline iscan/static/iscan/query-detail/query-detail.js]: Ok, wait, there's this here now in addition to the code above in `refreshPermissions`.  This block about permissions might be better moved to the `refreshPermissions` function?

**Comment by msonderegger** [comment]
@MichaelGoodale Wow, very nice!  Let's discuss today how to proceed + testing.

**Comment by msonderegger** [comment]
@MichaelGoodale once all fixes have been made and a version works on roquefort, could you check back in with me + James so we can start someone testing?

@james-tanner if you're willing I think it makes most sense for you to test -- which means trying to find bugs, but also just giving feedback on functionality as an actual user.

File: iscan/static/iscan/query-detail/query_detail.html [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
     <div class="col-md-9" style="height:100%">
         <div class="annotation-plot">
-            <annotation-plot data="utterance" selected-annotation="selectedAnnotation"></annotation-plot>
+            <annotation-plot data="utterance" selected-annotation="selectedAnnotation" can_edit="can_edit"></annotation-plot>
         </div>
 
         <div class="detail-plot">
             <waveform-plot data="waveform" begin="utterance.begin" end="utterance.end"
-                           selected-annotation="selectedAnnotation"></waveform-plot>
+                           selected-annotation="selectedAnnotation" can-edit="can_edit"></waveform-plot>
```
Comment (mmcauliffe) [inline iscan/static/iscan/query-detail/query_detail.html]: `can_edit` should be `can_edit`?

Comment (MichaelGoodale) [reply iscan/static/iscan/query-detail/query_detail.html]: No, it's to do with how directives handle the variables being moved into scope, I couldn't get can_edit to work as a name, only can-edit -> canEdit

**Pull request merged**